pkg/cmdutil/config.go

```diff
@@ -34,17 +34,21 @@ func GetFullConfig(defaultConfig *types.Config, clusterConfig *types.ClusterConf
 	return &ndf, &cc, nil
 }
 
-// ensureBootImageVersion patches the boot image tag as necessary based on the desired version
-// of Kubernetes. It returns the updated image string.
-func EnsureBootImageVersion(requestedVersion string, image string) string {
+// EnsureBootImageVersion appends an image tag consisting of the Kubernetes version ,if the image string does not currently have a tag.
+// It returns the updated image string.
+func EnsureBootImageVersion(kubeVersion string, image string) (string, error) {
 	// if the user already specified a tag at the end of the image, use that tag and return
-	parts := strings.Split(image, ":")
-	_, err := semver.NewVersion(parts[len(parts)-1])
-	if err == nil {
-		return image
+	imageTag, _, err := image2.GetTag(image)
+	if err != nil {
+		return image, err
+	}
+	// By default, the GetTag functions returns the latest tag if the image doesn't have a tag
+	// Below checks if the image actually has this latest tag or not
+	if imageTag == "latest" && !strings.HasSuffix(image, ":latest") {
```

Comment:
I think this check would be better placed inside the getTag function

Reply:
Or we can add a comment that explains getTag will return "latest" if no tag is found